specs/lightclient/beacon-chain.md

```diff
@@ -178,6 +340,105 @@ def process_block(state: BeaconState, block: BeaconBlock) -> None:
     process_sync_committee(state, block.body)
 ```
 
+#### New `process_attestation`
+
+*Note*: The function `process_attestation` is modified to do incentive accounting with epoch participation flags.
+
+```python
+def process_attestation(state: BeaconState, attestation: Attestation) -> None:
+    data = attestation.data
+    assert data.target.epoch in (get_previous_epoch(state), get_current_epoch(state))
+    assert data.target.epoch == compute_epoch_at_slot(data.slot)
+    assert data.slot + MIN_ATTESTATION_INCLUSION_DELAY <= state.slot <= data.slot + SLOTS_PER_EPOCH
+    assert data.index < get_committee_count_per_slot(state, data.target.epoch)
+
+    committee = get_beacon_committee(state, data.slot, data.index)
+    assert len(attestation.aggregation_bits) == len(committee)
+
+    if data.target.epoch == get_current_epoch(state):
+        epoch_participation = state.current_epoch_participation
+        justified_checkpoint = state.current_justified_checkpoint
+    else:
+        epoch_participation = state.previous_epoch_participation
+        justified_checkpoint = state.previous_justified_checkpoint
+
+    # Matching roots
+    is_matching_head = data.beacon_block_root == get_block_root_at_slot(state, data.slot)
+    is_matching_source = data.source == justified_checkpoint
+    is_matching_target = data.target.root == get_block_root(state, data.target.epoch)
+    assert is_matching_source
+
+    # Verify signature
+    assert is_valid_indexed_attestation(state, get_indexed_attestation(state, attestation))
+
+    # Participation flags
+    participation_flags = []
+    if is_matching_head and state.slot <= data.slot + MIN_ATTESTATION_INCLUSION_DELAY:
+        participation_flags.append(TIMELY_HEAD_FLAG)
+    if is_matching_source and state.slot <= data.slot + integer_squareroot(SLOTS_PER_EPOCH):
```

Comment:
@paulhauner
Why was `MIN_ATTESTATION_INCLUSION_DELAY` ingored?
Isn't `MIN_ATTESTATION_INCLUSION_DELAY = 1 < integer_squareroot(SLOTS_PER_EPOCH) = integer_squareroot(32) = 5` so it's covered?

Reply:
Yep, it works fine as-is. I just wanted to check because it seems like an easy place to say "we meant for the user to have a window of 5 slots but we accidentally gave them 4 because we forgot to subtract the delay".